<h1>Laboratorio 1 - Practicando Markdown y Git</h1>

<h2>Markdown</h2>

Markdown es un lenguaje de marcado ligero utilizado para dar formato a documentos de texto. Markdown utiliza una sintaxis simple, basada en signos de puntuación como #, * y > para dar formato al texto.

Ejemplos:
* Usar un símbolo # antes del texto crea un encabezado
* Usar asteriscos crea un énfasis 
* Usar corchetes angulares crea una cita en bloque

Esto hace que sea fácil escribir texto formateado de manera que sea legible fácilmente convertible a HTML.

Si se desea dar un formato de texto, se puede utilizar la documentación disponible en la pagina [markdownguide] (https://www.markdownguide.org/basic-syntax/)

Se utilizaran mas ejemplos a lo largo de este notebook.

<h2>Git</h2>

Git es un sistema de control de versiones distribuido que ayuda a los desarrolladores a gestionar los cambios en su código fuente y a colaborar en proyectos con multiples usuarios de manera eficiente.
Un sistema de control de versiones como Git, permite:

* Flujo de trabajo en ramas
* Desarrollar colaborativamente
* Mejor organización en el código
* Opciones de seguridad y recuperación

<h3>Aprendiendo algunos comandos de Git</h3>

Este ensayo servirá como ejemplo de algunos de los comandos básicos de Git.

* Comenzaremos utilizando el comando git init para indicar a Git que vamos a crear un nuevo repositorio.

In [6]:
%%cmd
git init

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git init
Reinitialized existing Git repository in C:/Users/LAPTOP-ERICK/Documents/Maestria Data Scientist/2023/Trimestre 1/Ciencia de datos con Python/Scripts Python/CDP_Markdown_and_git/.git/

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

* Creamos un archivo llamado Laboratorio1.ipynb y luego ejecutamos el comando git status.

In [7]:
%%cmd

git status

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git status
On branch master

No commits yet

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	Laboratorio1.ipynb

nothing added to commit but untracked files present (use "git add" to track)

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

El comando git status, nos muestra el estado actual de nuestro proyecto, en este caso nos dice que no tenemos commits aún y que debemos incluir el archivo Laboratorio1.ipynb para agregarlo a nuestra área de trabajo. Vamos a usar el comando que nos sugiere para agregar el archivo. (git add < file >)

In [9]:
%%cmd

git add Laboratorio1.ipynb

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git add Laboratorio1.ipynb

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

Ahora vamos a hacer nuestro primer commit, usaremos el comando "git commit -m 'texto' "

In [11]:
%%cmd

git commit -m "Primer commit de este ensayo"

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git commit -m "Primer commit de este ensayo"
[master (root-commit) 78267ab] Primer commit de este ensayo
 1 file changed, 38 insertions(+)
 create mode 100644 Laboratorio1.ipynb

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

Tenemos nuestro primer commit, el cual generó un codigo para identificarlo (100644)

Cabe mencionar que previamente debemos crear una cuenta en GitHub y crear un repositorio.
El repositorio creado para este ensayo se llama: CDP_Markdown_and_git

Pueden encontrarlo en el siguiente link (https://github.com/brandonfuentes97/CDP_Markdown_and_git.git)

Ahora vamos a integrar este notebook a nuestro repositorio en GitHub, para esto debemos utilizar los siguientes comandos:
* git remote add origin https://github.com/brandonfuentes97/CDP_Markdown_and_git.git
* git push -u origin master

In [12]:
%%cmd

git remote add origin https://github.com/brandonfuentes97/CDP_Markdown_and_git.git

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git remote add origin https://github.com/brandonfuentes97/CDP_Markdown_and_git.git

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

In [13]:
%%cmd

git push -u origin master

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git push -u origin master
branch 'master' set up to track 'origin/master'.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

To https://github.com/brandonfuentes97/CDP_Markdown_and_git.git
 * [new branch]      master -> master


Bien, ahora ya tenemos nuestro repositorio con el primer commit en GitHub.

Vamos a ejecutar nuevamente el comando git status, para revisar el estado actual de nuestro proyecto.

In [17]:
%%cmd

git status

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   Laboratorio1.ipynb

no changes added to commit (use "git add" and/or "git commit -a")

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

Nos indica que nuestro notebook Laboratorio1.ipynb tiene modificaciones, ya que hasta este momento se guardó el archivo (presionamos ctrl+s). Debemos ejecutar nuevamente el comando git add para volver a indicarle a git que tome estos cambios dentro de nuestro proyecto.

In [18]:
%%cmd

git add Laboratorio1.ipynb

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git add Laboratorio1.ipynb

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

Y hacemos nuestro segundo commit

In [19]:
%%cmd

git commit -m "Segundo commit! Vamos bien"

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git commit -m "Segundo commit! Vamos bien"
[master b45c250] Segundo commit! Vamos bien
 1 file changed, 381 insertions(+), 2 deletions(-)

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

Muy bien, ya tenemos nuestro segundo commit, pero aún no lo vemos reflejado en nuestro repositorio de GitHub, por lo que vamos a ejecutar nuevamente el comando git push -u origin master

In [20]:
%%cmd

git push -u origin master

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git push -u origin master
branch 'master' set up to track 'origin/master'.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

To https://github.com/brandonfuentes97/CDP_Markdown_and_git.git
   78267ab..b45c250  master -> master


Tomamos una captura de los commit en el repositorio de GitHub y la vamos a mostrar a continuación:

<img src="img-commit.png">

In [22]:
%%cmd

git status

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   Laboratorio1.ipynb

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	img-commit.png

no changes added to commit (use "git add" and/or "git commit -a")

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

Como agregamos una imagen dentro de la carpeta raiz de este notebook, git lo reconoce y nos avisa que debemos ejecutar el comando git add nuevamente. Lo vamos a ejecutar, pero en esta ocasión no colocaremos el nombre del archivo, sino un punto, para que tome cualquier archivo nuevo.

In [25]:
%%cmd 

git add .

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git add .

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

In [26]:
%%cmd

git commit -m "Tercer commit, agregamos una imagen!"

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git commit -m "Tercer commit, agregamos una imagen!"
[master 4197a28] Tercer commit, agregamos una imagen!
 2 files changed, 188 insertions(+), 3 deletions(-)
 create mode 100644 img-commit.png

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

In [27]:
%%cmd

git push -u origin master

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git push -u origin master
branch 'master' set up to track 'origin/master'.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

To https://github.com/brandonfuentes97/CDP_Markdown_and_git.git
   b45c250..4197a28  master -> master


Ahora veremos nuevos comandos para utilizar ramas en nuestro desarrollo.

Por ejemplo:

* git branch : Nos muestra las ramas existentes
* git branch name_branch : crea una rama con el nombre que le indiquemos
* git checkout name_branch : nos cambia a la rama que indiquemos

Para este ejemplo, crearemos la rama llamada "dev" y nos moveremos a ella.

La ramificaciones se utilizan para tener una copia del proyecto principal y trabajar sobre ella sin alteral la rama principal. Como nons cambiaremos a la rama dev, todos los cambios que se relicen solo afectarán a esta rama.

In [28]:
%%cmd

git branch

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git branch
* master

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

In [29]:
%%cmd

git branch dev

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git branch dev

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

In [30]:
%%cmd

git branch

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git branch
  dev
* master

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

In [33]:
%%cmd

git checkout dev

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git checkout dev
M	Laboratorio1.ipynb

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

Switched to branch 'dev'


**Este texto solo está incluida en la rama dev**

In [34]:
%%cmd 

git status

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git status
On branch dev
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   Laboratorio1.ipynb

no changes added to commit (use "git add" and/or "git commit -a")

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

In [35]:
%%cmd

git add .

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git add .

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

In [36]:
%%cmd

git commit -m "Este commit corresponde a la rama dev"
git push -u origin dev

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git commit -m "Este commit corresponde a la rama dev"
[dev 299bb9f] Este commit corresponde a la rama dev
 2 files changed, 226 insertions(+), 3 deletions(-)
 create mode 100644 Laboratorio1_branch_dev.ipynb

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git push -u origin dev
branch 'dev' set up to track 'origin/dev'.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

remote: 
remote: Create a pull request for 'dev' on GitHub by visiting:        
remote:      https://github.com/brandonfuentes97/CDP_Markdown_and_git/pull/new/dev        
remote: 
To https://github.com/brandonfuentes97/CDP_Markdown_and_git.git
 * [new branch]      dev -> dev


In [38]:
%%cmd

git checkout master

Microsoft Windows [Version 10.0.22621.1105]
(c) Microsoft Corporation. All rights reserved.

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>
(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>git checkout master

(base) c:\Users\LAPTOP-ERICK\Documents\Maestria Data Scientist\2023\Trimestre 1\Ciencia de datos con Python\Scripts Python\CDP_Markdown_and_git>

error: Your local changes to the following files would be overwritten by checkout:
	Laboratorio1.ipynb
Please commit your changes or stash them before you switch branches.
Aborting
